File: backend-springboot/src/main/java/com/javeriana/pontimovil/ponti_movil/dto/pasajero/RutaDTO.java
package com.javeriana.pontimovil.ponti_movil.dto.pasajero;

import lombok.Getter;
import lombok.Setter;
import java.io.Serializable;
import lombok.*;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@EqualsAndHashCode
@NoArgsConstructor
public class RutaDTO implements Serializable {
    String codigo;
    HorarioDTO horario;
    List<EstacionDTO> estaciones;
    List<BusDTO> buses;
    List<ConductorDTO> conductores;
}

File: frontend-angular/src/app/dto/gestionar-rutas/ruta/ruta-dto.ts
import {HorarioDTO} from "./horario-dto";
import {EstacionDto} from "./estacion-dto";
import {BusDTO} from "./bus-dto";
import {ConductorDTO} from "./conductor-dto";

export class RutaDto {
  constructor(
    public id: string,
    public codigo: string,
    public horario: HorarioDTO,
    public estaciones: EstacionDto[],
    public buses: BusDTO[],
    public conductores: ConductorDTO[]
  ) {}
}
